src/core/reactor.cc

```diff
@@ -1172,7 +1172,10 @@ cpu_stall_detector_posix_timer::cpu_stall_detector_posix_timer(cpu_stall_detecto
     struct sigevent sev = {};
     sev.sigev_notify = SIGEV_THREAD_ID;
     sev.sigev_signo = signal_number();
-    sev._sigev_un._tid = syscall(SYS_gettid);
+#ifndef sigev_notify_thread_id
+#define sigev_notify_thread_id _sigev_un._tid
+#endif
```

Comment:
a little bit background in the commit message would be great, something like

> _sigev_un._tid is an internal glibc field and is not available on
musl libc. 
> see https://sourceware.org/bugzilla/show_bug.cgi?id=27417